addon/store.js

```diff
@@ -46,45 +46,39 @@ var Store = Ember.Object.extend({
       this.set("array", {});
     },
     clear(type) {
-        if(type === undefined) {
+        if (type === undefined) {
             this.reset();
         }
+
         delete this.get("identityMap")[type];
         arrayForType(type, this).clear();
-
-        run.once(() => {
-            this.scheduleUpdate(type);
-        });
+        this.scheduleUpdate(type);
     },
     remove(type, id) {
         var record = this._findById(type, id);
         if (record) {
             delete this.get("identityMap")[type][record.id];
             arrayForType(type, this).removeObject(record);
-
-            run.once(() => {
-                this.scheduleUpdate(type);
-            });
+            this.scheduleUpdate(type);
         }
     },
     push(type, data) {
         var record = this._findById(type, data.id);
+
         if (record) {
             setProperties(record, data);
         } else {
             record = buildRecord(type, data, this);
         }
 
-        run.once(() => {
-            this.scheduleUpdate(type);
-        });
+        this.scheduleUpdate(type);
 
         return record;
     },
     scheduleUpdate(type) {
         var recompute = this.get("recompute");
-        recompute.push(type);
-        run.scheduleOnce("actions", this, "updateFilters");
+        recompute.addObject(type);
```

Comment:
Ensures only unique values are pushed to the array, so we don't schedule the same type multiple times to be recomputed